### ISSUE TITLE
Scaffolder Create Component Frontend Wizard

### ISSUE THREAD

**Issue by benjdlambert** [spotify/backstage#1189]
We need a way to be able to create components using the Scaffolder in the frontend.
Parameters needed for the scaffolder should live with the template, and then displayed in the frontend wizard as inputs.

It would be a good to use JSON Forms schema to be able to declare the input variables for each template, and then submit these variables to the backend when creating a new component for scaffolding.
 
## Possible Implementation
I put together a quick demo about using JSON forms schema and `mui` here which is a POC on how it could be stored in the template and then rendered in React world.

https://github.com/spotify/backstage/compare/blam/json-forms

Depends on #1108 
Related to #345

**Comment by benjdlambert** [comment]
Create a new ticket to copy some of the complex internal field components like `reposelecetion` and. other async data loading.

**Comment by benjdlambert** [comment]
Superceeded by #1299. Cloising
